src/helpers/validateLink.ts

```diff
@@ -0,0 +1,12 @@
+export const validateLink = (linkValue:string, setErrorMsg:(value:string) => void ) => {
+  const isLinkValue = /^(https?):\/\/[^\s/$.?#].[^\s]*$/i.test(linkValue);
+
+  if(linkValue === '') {
+    setErrorMsg('Please add a link');
+  } else if (!isLinkValue) {
+    setErrorMsg('Invalid URL submitted');
+  } else {
+    setErrorMsg('');
+  }
```

Comment:
Mandar el setError sería un poco dificil si quisieras testear este método de manera aislada. Otra manera de manejar esto sería la siguiente:

```suggestion
  if(linkValue === '') {
    'Please add a link';
  } else if (!isLinkValue) {
    return 'Invalid URL submitted'
  }
  return ""
```

La manera de utilizarlo sería:

```tsx
const linkErrorMessage = validateLink(link);

if(hasLinkValidationErrors) setErrorMsg(linkErrorMessage)
else
  // Procedes a generar la URL
```